app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -0,0 +1,22 @@
+module Katello
+  class Api::V2::UpstreamSubscriptionsController < Api::V2::ApiController
+    before_action :find_organization
+
+    resource_description do
+      description "Subscriptions management on the CDN."
+      api_version 'v2'
+    end
+
+    api :GET, "/organizations/:organization_id/upstream_subscriptions", N_("List available subscriptions from the CDN")
+    param :organization_id, :number, :desc => N_("Organization ID"), :required => true
+    def index
+      render json: cp_resource.get_for_owner
```

Comment:
This method blows up when the org doesn't have a manifest imported. I think we should handle it more gracefully:

```
2018-02-26T15:39:49 5d4145dc [app] [E] NoMethodError: undefined method `[]' for nil:NilClass
 | /home/vagrant/katello/app/lib/katello/resources/candlepin.rb:149:in `upstream_api_uri'
 | /home/vagrant/katello/app/lib/katello/resources/candlepin.rb:153:in `site'
 | /home/vagrant/katello/app/lib/katello/resources/candlepin.rb:117:in `resource'
 | /home/vagrant/katello/app/lib/katello/resources/candlepin.rb:165:in `upstream_owner_id'
 | /home/vagrant/katello/app/lib/katello/resources/candlepin/pool.rb:53:in `get_for_owner'
 | /home/vagrant/katello/app/controllers/katello/api/v2/upstream_subscriptions_controller.rb:13:in `index'
```

Reply:
Good catch! Thanks for breaking! :)

Reply:
@jturel updated.

```
RuntimeError: Current organization has no manifest imported.
```

Reply:
I would also recommend against rendering pure json back to the client.  We've done this in the past and it has always bitten us, for the following reasons:

* inconsistent data types (here you have candlepin based param naming (productName instead of product_name).  
* if candlepin changes some value, we propagate that change to the client (unlikely in this case)
* harder to control what gets sent to the client (i.e. we may want to reduce what it returned to improve performance).

Instead, i would recommend we populate some model object that is a simple holder of attributes.  Then we can render this as rabl just like all the other api controllers.